Use .cozmars directory

anjok opened the pull request:
- should use a directory instead of a bunch of files
- fix typo in README.md
- remove comment from conf.json

anjok commented:
Actually, please don't pull this yet. I can't use different branches for the different issues as I don't know your schedule and they almost all overlap.

So what I did here:

- added gettext + locale
- moved all .tmpl to .html, as they are easier to view in the finder preview
- fixed all Chinese text to google-translated gettext messages

The next step is to generate the locales/base.dot file and add the various translations.

Please don't pull this yet, as I can't test any of this due of a lack of a configured raspi. I still need some parts.

But if possible, don't create conflicts for me yet...

anjok commented:
Unfortunately you would need to regenerate the Chinese text from the to-be-created gettext files. Since I have no idea what the characters are, I would just create a mess and it would take ages.

anjok commented:
@hyansuper  ok, localisation seems to work now. It would be great if you could take a look.

hyansuper commented:
thank you for the effort, I am looking at it.
I have not used gettext before, may take a while

anjok commented:
Please re-fetch the PR, I squashed the commits and rebased on your latest code.

Note that it's *very* hard for me to merge incoming changes of yours into my branch as I have no idea what your code actually spells... so please decide soon if you want to do the localisation or not. If not, I have to keep my own fork and won't be able to contribute.

I was working with this here:

https://phrase.com/blog/posts/translate-python-gnu-gettext/

it's basically:

`
cd rcute_cozmars_server
(wherever this is)/pygettext.py -d base -o locales/base.pot *.py *.html
`

Then generate loc files for each non-english language and fill in the blanks. It's really easy to use.

You'd need to proofread the English (as they are the keys, they need to be pretty good) and fill in the generated Chinese.

hyansuper commented on rcute_cozmars_server/static/servo.html:
```diff
@@ -1,65 +1,76 @@
 </script>
-<h1>舵机校准</h1>
-<p><!-- 出厂时舵机已经调试好，一般不需要再进行校准<br> -->只有在 <a href='test'>测试</a> 中舵机无法到达指定位置时才需要校准</p><p style='background-color:yellow;text-align:left;color:red;padding:10px;'>警告: 点击保存后将无法恢复原设置！</p>
-<h2>头部&nbsp;&nbsp;&nbsp;&nbsp;<span id='head_val'>0°</span></h2>
```
did you missd line 55~56, they are for head servo calibration

hyansuper commented on rcute_cozmars_server/util.py:
```diff
@@ -1,5 +1,27 @@
+    return _gettext(match)
+
+def parsed_template(name, **kwargs):
+    with open(util.static("{}.html".format(name))) as file:
```
"util.static" -> just "static"

hyansuper commented on rcute_cozmars_server/util.py:
```diff
@@ -1,5 +1,27 @@
 from PIL import Image, ImageFont, ImageDraw
+import gettext
+import locale
+
```
import re

hyansuper replied:
import re

hyansuper commented on rcute_cozmars_server/static/test.html:
```diff
@@ -1,30 +1,54 @@
+	<button onclick='robot.head.angle(-30)'>{_("Head down")}</button>
+	<button onclick='robot.lift.height(1)'>{_("Lift up")}</button>
+	<button onclick='robot.lift.height(0)'>{_("Lift down")}</button>
+<h3>{_("Motor test")}  <input id='sec' type='number' size='1' value='2'/>{_("Speed")}</h3>
```
"Speed" -> "second"

hyansuper replied:
_("Speed") => _("seconds")

hyansuper commented on requirements.txt:
```diff
@@ -4,6 +4,8 @@ gpiozero
 picamera
 sounddevice
 sanic
+gettext
```
no need to require gettext and locale, they are builtin with python3, they can't be installed individually

hyansuper replied:
gettext and locale is builtin with python3. don't need to require gettext and locale, they can't be installed individually

hyansuper commented on rcute_cozmars_server/__main__.py:
```diff
@@ -1,12 +1,14 @@
 import asyncio
 import sanic
-
+import re
```
move this line to file util.py

hyansuper commented on rcute_cozmars_server/static/index.html:
```diff
@@ -0,0 +1,58 @@
+		<h2>{_("Settings")}</h2>
+		<a href='wifi'><button>{_("WiFi Settings")}</button></a>
+		<a href='test'><button>{_("Functional Test")}</button></a>
+		<a href='servo'> <button{_("Servo calibration")}</button></a><a href='motor'><button>{_("Motor calibration")}</button></a>
```
"<button" -> left a ">" sign

hyansuper commented on rcute_cozmars_server/static/motor.html:
```diff
@@ -1,66 +1,85 @@
-	function reset_config() {
-		return fetch('http://'+location.host+'/conf').then(j=>j.json()).then(c=>{
+	function reset_config() {{
+		return fetch('http://'+location.host+'/config').then(j=>j.json()).then(c=>{{
```
config -> conf

hyansuper commented on rcute_cozmars_server/static/servo.html:
```diff
@@ -1,65 +1,76 @@
-	function reset_config() {
-		return fetch('http://'+location.host+'/conf').then(j=>j.json()).then(c=>{
+	function reset_config() {{
+		return fetch('http://'+location.host+'/config').then(j=>j.json()).then(c=>{{
```
config -> conf

hyansuper commented on rcute_cozmars_server/static/test.html:
```diff
@@ -1,30 +1,54 @@
 <script src="https://cdn.jsdelivr.net/npm/@ygoe/msgpack@1.0.2/msgpack.min.js"></script>
 <script src="https://cdn.jsdelivr.net/gh/hyansuper/wsmprpc/js/client.min.js"></script>
-<script src="https://cdn.jsdelivr.net/gh/hyansuper/rcute-cozmars-server/rcute_cozmars_server/static/cozmars.min.js"></script>
+<script src="https://cdn.jsdelivr.net/gh/hyansuper/rcute-cozmars-server/server/static/cozmars.min.js"></script>
```
this line should not change

hyansuper reviewed (changes requested):
just a few

hyansuper commented:
sorry I made some duplicated change requests

anjok commented:
ok, will fix and push later

anjok commented:
@hyansuper please review again.

anjok commented:
also, see:

https://stackoverflow.com/questions/27214065/how-to-docstring-in-python-for-multiple-languages

for the doctoring comments.

hyansuper commented:
@anjok  did you have to hack pygettext.py in order to handle .html? 
I think this command also works `pygettext -d base -o locales/base.pot *.py static/*.html `
but a few english aren't extracted in html, especially those in javascript, maybe it has to do with "quote in quote in quoat...". I have to manually type into .pot file

anjok commented:
Oh ok. I didn't verify it got everything. Worst case one need to put the error strings in a server side duct
